Add flux event weight

Ooohu opened the pull request:
Add a BNBFlux folder for flux uncertainty evaluation.

cjbackhouse commented on sbncode/FluxReader/DK2NuInterface.cxx:
```diff
@@ -0,0 +1,291 @@
+      std::vector<double> rotmat=ps.get<std::vector<double> >("rotmatrix");
+      TVector3 newX,newY,newZ;
+      if (rotmat.size()==9) {
+	std::cout<<"Matrix defined with new axis values"<<std::endl;
```
In general, I think all printouts should be prefixed with the name of the file that printed them, at least. Otherwise some of these are quite cryptic, and it will be very difficult to track down what they actually mean. Some people would insist on using the proper message facility.

cjbackhouse commented on sbncode/FluxReader/DK2NuInterface.cxx:
```diff
@@ -0,0 +1,291 @@
+      fBeamRotInv = fBeamRot.Inverse();
+    } catch (std::exception& e) {
+      std::cout<<e.what()<<std::endl;
+      std::cout<<"These are not numbers"<<std::endl;
```
I think the try/catch can just be around the `ps.get()`. If something goes wrong any further through this, it's a different cause.
And you should definitely call `abort()` after this happens. If the matrix is not configured properly, it's no good printing a vague message to the screen and then continuing with random element values. Similar comments for various printouts further down.

cjbackhouse commented on sbncode/FluxReader/DK2NuInterface.cxx:
```diff
@@ -0,0 +1,291 @@
+    double tiltwgt = p3beam.Unit().Dot( fWindowNormal );
+    wgt*=tiltwgt;
+    //weight for the window area and divide by pi (since wgt returned by calcEnuWgt function is flux/(pi*m^2)
+    wgt*=fWindowArea/3.14159;
```
`TMath::Pi()` or `M_PI`. And I am pretty confused by the unit convention that requires this....

cjbackhouse commented on sbncode/FluxReader/FluxGeoFilter_module.cc:
```diff
@@ -0,0 +1,102 @@
+{
+  std::cout<<"Configuring flux filter."<<std::endl;
+  std::vector<std::string> tmp=p.get<std::vector<std::string> >("volumes");
+  fVolFlux.insert(tmp.begin(),tmp.end());
```
Why not just `fVolFlux = p.get<...`?

cjbackhouse commented on sbncode/FluxReader/FluxGeoFilter_module.cc:
```diff
@@ -0,0 +1,102 @@
+      rgeo->FindNextBoundary();
+      node=gGeoManager->Step();
+      if (fVolFlux.find(volname)!=fVolFlux.end()) {
+        result=true;
```
Personally I would prefer to just `return true;` here and get rid of the `result` variable.

cjbackhouse commented on sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.cxx:
```diff
@@ -0,0 +1,351 @@
+
+
+namespace sbn {
+	namespace evwgh {
```
Previous files have been indented with 2 spaces, which is the larsoft standard. This one looks to be tabs.

cjbackhouse commented on sbncode/SBNEventWeight/Calculators/BNBFlux/PrimaryHadronSanfordWangWeightCalc.cxx:
```diff
@@ -0,0 +1,153 @@
+           
+      double weight = 1; 
+
+      if(RW < 0 || CV < 0){//dont bother this; if this happens, the weight would be skipped...
```
I think this file is a mix of tabs and spaces? It is certainly all messed up in the github viewer.

cjbackhouse commented:
As you can see, I have made a bunch of annoying superficial style comments (some/all of them may have been inherited from the uboone originals?), but I am pretty out of my depth on the actual calculations going on here. Probably an actual flux expert should look this over. I will try and review a sample output file to make sure this will be ingestible by CAFMaker.

cjbackhouse commented on sbncode/CMakeLists.txt:
```diff
@@ -14,4 +14,9 @@ add_subdirectory(Metadata)
 add_subdirectory(FlashMatch)
 add_subdirectory(LArRecoProducer)
 add_subdirectory(TPCReco)
+<<<<<<< HEAD
```
merge conflict here...

wesketchum commented:
@Ooohu if this is ready to merge, can we move this from a draft to a real PR?

One essential update is that we need histogram files in an available sbndata package, which also needs to be distributed. Need to followup with @miquelnebot on status of that.

jzennamo commented on sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.cxx:
```diff
@@ -0,0 +1,352 @@
+          if(validC){//load random numbers based on FitCov
+            //{2*multisim vector}
+            //{{Ncols() elements}} <-- feed these amount everytime;
+            for( int index = 1; index < 2*(FitCov->GetNcols()); index ++){
+              fParameterSet.Sample(engine);//load it 2*<number_of_multisims> times
+            }
```
Why do you have to load this so many times? Maybe you could explain to me with this is doing specifically?

Ooohu replied:
Yes, this is to avoid segmentation fault. in case a random number causes a bad cross-section, i.e. negative value, then we use another (next) random number to estimate a new cross-section. We need sufficient random number to go through this process.

See the loop in the following section:
https://github.com/Ooohu/sbncode/blob/d324b83a33b9988b7091afd7624b4e0bb92b1e51/sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.cxx#L290

This loop repeats until the size of `<vector> weights` is large enough, and `i++` always runs (line 290); In line 312, the size of `weights` grows only if a good `test_weight` is calculated, otherwise, the loop repeated with extra `i++`. Here `Vrandom[i]` at line 296 may go beyond the `Nuni` (number of multisims).

jzennamo replied:
OK, so `fParameterSet.Sample(engine);` generates a new random number, and stores in in a map as part of fParameterSet. Got it. Thanks

jzennamo commented on sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.cxx:
```diff
@@ -0,0 +1,352 @@
+      //or do the above 3 lines in one line
+      auto const& mclist = *e.getValidHandle<std::vector<simb::MCTruth>>(fGenieModuleLabel);
+
+      // If mo neutrinos in this event, gives 0 weight;
```
"If _no_ neutrinos", typo fix

jzennamo commented on sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.cxx:
```diff
@@ -0,0 +1,352 @@
+        }
+
+        // Collect neutrino energy
+        double enu= mclist[inu].GetNeutrino().Nu().E();
```
Instead of using MCTruth for this could you use:

`double enu = fluxlist[inu].fnenergy`

Then you could drop all references to `MCTruth`

Ooohu replied:
Sounds good! I just did not check it yet, but yes, if this works!

Ooohu replied:
It is `fluxlist[inu].fnenergyn` and I have applied the changes.

jzennamo commented on sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.cxx:
```diff
@@ -0,0 +1,352 @@
+          for (unsigned int i = 0; int(weights.size()) < NUni; i++) {//if all weights are 1, no need to calculate weights;
+            std::pair<bool, double> test_weight;
+
+            std::vector< float > Vrandom = (fParameterSet.fParameterMap.begin())->second;//vector of random #
+            std::vector< float > subVrandom;//sub-vector of random numbers;
```
why make these `std::vector <float>` instead of `std::vector<double>`

Ooohu replied:
Same as above, use float to reduce the output root size.

jzennamo commented on sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.h:
```diff
@@ -0,0 +1,91 @@
+        //GetWeight() returns the final weights as a vector
+        //  each weight evaluaed by *WeightCalc() function;
+        //inu - the ith parameter.
+        std::vector<float> GetWeight(art::Event& e, size_t inu) override;
```
offhand I feel like weights should be `double`s, any reason we think that `float`s are OK?

Ooohu replied:
A quick reference: https://hackr.io/blog/float-vs-double

The table shows that..


|Floating point type | Memory requirement | Range|
|-- | -- | --|
|Float | 4 bytes | ±3.40282347E+38F i.e. 6-7 significant digits|
|Double | 8 bytes | ±1.79769313486231570E+308 i.e. 15-16 significant digits|

It looks like 6 significant digits are sufficient for weights?

jzennamo commented on sbncode/SBNEventWeight/jobs/flux/eventweight_flux_sbn.fcl:
```diff
@@ -0,0 +1,232 @@
+
+  piminus: {
+    type: Flux
+    random_seed: 3
```
how random are these seeds?

Ooohu replied:
This seed is what was used in MicroBooNE, and it uses the same  `CLHEP/Random/RandGaussQ.h` to generate random numbers.

jzennamo reviewed (commented):
This is fantastic work! Thank you so much for spearheading this!

Ooohu commented on sbncode/SBNEventWeight/Base/SmearingUtils.cxx:
```diff
@@ -86,7 +86,7 @@ std::vector<std::vector<double> > MultiGaussianSmearing(
     std::vector<double> const& centralValue,
     TMatrixD* const& inputCovarianceMatrix,
-    std::vector< double > rand) {
+    std::vector< float > rand) {
```
It comes from the sbnobj:
https://github.com/SBNSoftware/sbnobj/blob/develop/sbnobj/Common/SBNEventWeight/EventWeightParameterSet.cxx#L64

It was changed because saving data in float can reduce the file size, where double has 8 bytes and float has 4 bytes.

jzennamo replied:
Have you studied if there is a difference between the results if you use floats or doubles? If not could we change these to doubles for now and then study if having floats change the answer?

Ooohu commented on sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.cxx:
```diff
@@ -0,0 +1,352 @@
+        CLHEP::HepRandomEngine& engine) {
+      std::cout<<"SBNEventWeight Flux: configure "<< GetName() << std::endl;
+
+      fGenieModuleLabel = p.get<std::string>("genie_module_label");//use this label to get MC*Handle
```
Yes! Also needs to update the parameter names in the FHiCL file.

jzennamo commented on sbncode/CMakeLists.txt:
```diff
@@ -4,18 +4,18 @@ cet_find_library( BOOST_DATE_TIME     NAMES boost_date_time     PATHS ENV BOOST_
 add_subdirectory(FluxReader)
-add_subdirectory(EventGenerator)
-add_subdirectory(PID)
+#add_subdirectory(EventGenerator)
```
Did you mean to comment these out? These directories seem needed

Ooohu replied:
no, I just need to remove them from compile error. I will change it back.

Ooohu replied:
The compiling errors came from my old sbncode version (v09_24_00); I use v09_29_00 and it works fine.

jzennamo commented on sbncode/SBNEventWeight/Calculators/BNBFlux/FluxCalcPrep.cxx:
```diff
@@ -250,8 +251,9 @@ namespace sbn {
 
-        // Collect neutrino energy
-        double enu= mclist[inu].GetNeutrino().Nu().E();
+        // Collect neutrino energy; mclist is replaced with fluxlist.
+//        double enu= mclist[inu].GetNeutrino().Nu().E();
+        double enu= fluxlist[inu].fnenergyn;
```
Just to verify, did you make sure that the units are consistent between `mclist[inu].GetNeutrino().Nu().E()` and `fluxlist[inu].fnenergyn`?

Ooohu replied:
Yes I got the same number from that variable.

jzennamo reviewed (approved):
Excellent! Looking forward to playing with these in the next release

FNALbuild commented:
CI build for LArSoft on slf7 for e20:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/12704&builds=)

FNALbuild commented:
CI build for LArSoft on slf7 for c7:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/12703&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for e20:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/12704&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c7:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/12703&builds=)

FNALbuild commented:
CI build for ICARUS on slf7 for e20:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2622&builds=)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/12704&builds=)

FNALbuild commented:
CI build for SBND on slf7 for e20:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/7271&builds=)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/12704&builds=)

FNALbuild commented:
CI build for ICARUS on slf7 for c7:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2623&builds=)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/12703&builds=)

FNALbuild commented:
CI build for SBND on slf7 for c7:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/7272&builds=)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/12703&builds=)

FNALbuild commented:
:x: CI build for ICARUS Failed at phase build ICARUS on slf7 for e20:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2622&builds=)

:rotating_light: For more details about the failed phase, check the [build ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/2622&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=build&buildtype=slf7%20e20:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/12704&builds=)

FNALbuild commented:
:x: CI build for ICARUS Failed at phase build ICARUS on slf7 for c7:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2623&builds=)

:rotating_light: For more details about the failed phase, check the [build ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/2623&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=build&buildtype=slf7%20c7:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/12703&builds=)

FNALbuild commented:
:x: CI build for SBND Failed at phase build SBND on slf7 for e20:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/7271&builds=)

:rotating_light: For more details about the failed phase, check the [build SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/7271&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=build&buildtype=slf7%20e20:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/12704&builds=)

FNALbuild commented:
:x: CI build for SBND Failed at phase build SBND on slf7 for c7:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/7272&builds=)

:rotating_light: For more details about the failed phase, check the [build SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/7272&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=build&buildtype=slf7%20c7:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/12703&builds=)

cjbackhouse commented:
The errors start with
```12: 'sbnobj/ups/product_deps' -> 'sbnobj/ups/product_deps.bck'
13: ERROR: failed to find existing version for /scratch/workspace/icarus_ci/label_exp/SLF7/label_exp2/swarm/ICARUS/srcs/sbnobj/```

or
```10: 'sbnobj/ups/product_deps' -> 'sbnobj/ups/product_deps.bck'
11: ERROR: no version given```

I'm not sure that is anything to do with the code here? Maybe a more general CI problem?

henrylay97 commented:
Ah, admittedly confusing the error @cjbackhouse has mentioned isn't strictly an error and won't have been fatal. What that is referring to is that it can't find version numbers in the `product_deps` file any more (due to the mrb change). I believe Vito is looking into this but its non-fatal and the build can still succeed. 

The fatal error begins on line 795 of this [log](https://dbweb8.fnal.gov:8443/LarCI/app/ns:sbnd/storage/docs/2021/09/29/buildSBND%230mvEMQC.log).

cjbackhouse commented:
ok, this is a handful of missing includes.

The log I looked at didn't get this far, and just had the missing version messages. Maybe this is a separate issue, which is that I never know how to find the actual build I need to look at from the list of all the recent ones.

henrylay97 commented:
> x CI build for SBND Failed at phase build SBND on slf7 for c7:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/7272&builds=)
> 
> rotating_light For more details about the failed phase, check the [build SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/7272&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=build&buildtype=slf7%20c7:prof)
> 
> parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/12703&builds=)

So these messages from FNALbuild give the links to the relevant logs :)

henrylay97 commented:
Ahhh no you weren't looking at the wrong ones. ICARUS have yet to cut a release inheriting the recent art, mrb and cet changes. So their builds will fail regardless. The SBND one is required to actually see the failure. I hadn't spotted that earlier, apologies.

To sum up my earlier message is still accurate in terms of the fixes required before merge :)

cjbackhouse commented:
What I had clicked on before was the "Details" link next to a particular build under "Some checks were not succesful". That takes you to the top-level list of builds, not the specific build like the actual FNALbuild posts do.

Ooohu commented:
> @Ooohu Any news on addressing the issues in the CI tests?

I think I still need to solve the following error, which comes up when using v9_33_00.

> Ah, admittedly confusing, the error @cjbackhouse has mentioned isn't strictly an error and won't have been fatal. What that is referring to is that it can't find version numbers in the `product_deps` file any more (due to the mrb change). I believe Vito is looking into this but its non-fatal and the build can still succeed.
> 
> The fatal error begins on line 795 of this [log](https://dbweb8.fnal.gov:8443/LarCI/app/ns:sbnd/storage/docs/2021/09/29/buildSBND%230mvEMQC.log).

cjbackhouse commented:
All the stuff at the start is some CI-internal problem and not your fault. You do need to add a `#include` for fhicl::ParameterSet which seems to be the cause of the other problems.

FNALbuild commented:
CI build for ICARUS on slf7 for e20:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2778&builds=)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/13046&builds=)

FNALbuild commented:
:x: CI build for SBND Failed at phase ci_tests SBND on slf7 for c7:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/7469&builds=)

:rotating_light: For more details about the failed phase, check the [ci_tests SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/7469&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=ci_tests&buildtype=slf7%20c7:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/13047&builds=)

FNALbuild commented:
:x: CI build for ICARUS Failed at phase ci_tests ICARUS on slf7 for c7:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2779&builds=)

:rotating_light: For more details about the failed phase, check the [ci_tests ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/2779&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=ci_tests&buildtype=slf7%20c7:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/13047&builds=)

FNALbuild commented:
:warning: CI build for ICARUS Warning at phase ci_tests ICARUS on slf7 for e20:prof - ignored warnings for build -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2796&builds=)

:rotating_light: For more details about the warning phase, check the [ci_tests ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/2796&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=ci_tests&buildtype=slf7%20e20:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/13078&builds=)

FNALbuild commented:
:warning: CI build for ICARUS Warning at phase ci_tests ICARUS on slf7 for c7:prof - ignored warnings for build -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2795&builds=)

:rotating_light: For more details about the warning phase, check the [ci_tests ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/2795&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=ci_tests&buildtype=slf7%20c7:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/13077&builds=)

marcodeltutto commented:
Hi @Ooohu @miquelnebot, this PR needed some `beamData/` histogram files moved to `sbnd_data`, but I can't find them in `sbnd_data` `v01_10_00` and the code fails when running the flux reweighing. Am I missing something or were the files never copied? Thank you!

marcodeltutto commented:
Sorry I confused sbn with sbnd_data! 

If I only setup `sbndcode`, I don't get `sbndata` setup, and so my job did not find the files. Shall we have `sbndcode` also depend on `sbndata`? Thanks!

Ooohu commented:
> Sorry I confused sbn with sbnd_data!
> 
> If I only setup `sbndcode`, I don't get `sbndata` setup, and so my job did not find the files. Shall we have `sbndcode` also depend on `sbndata`? Thanks!

I think we will do it in a new PR.

marcodeltutto commented:
Thanks @miquelnebot!
